src/Serilog/Settings/KeyValuePairs/SettingValueConversions.cs

```diff
@@ -31,11 +32,16 @@ class SettingValueConversions
         {
                 { typeof(Uri), s => new Uri(s) },
                 { typeof(TimeSpan), s => TimeSpan.Parse(s) },
-                { typeof(Type), s => Type.GetType(s, throwOnError: true) },
+                { typeof(Type), s => Type.GetType(s, throwOnError: true)! },
             };
 
-        public static object ConvertToType(string value, Type toType)
+        public static object? ConvertToType(string? value, Type toType)
         {
+            if (value == null)
+            {
+                return null;
+            }
```

Comment:
I think this will break; if I ask for `ConvertToType(null, DateTimeKind)`, `null` is not a valid return value (it'll just trigger a failure further up the stack, when we try to use `null` as a `DateTimeKind`).